### PR TITLE
ansible-test - Fix consistency of managed venvs.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

ansible-test - Fix consistency of managed venvs.

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

ansible-test

### CODE REVIEW

File: test/lib/ansible_test/_internal/python_requirements.py [hunk 1]
```diff
@@ -207,15 +231,20 @@ def collect_requirements(
...
-    if minimize:
-        # In some environments pkg_resources is installed as a separate pip package which needs to be removed.
-        # For example, using Python 3.8 on Ubuntu 18.04 a virtualenv is created with only pip and setuptools.
-        # However, a venv is created with an additional pkg-resources package which is independent of setuptools.
-        # Making sure pkg-resources is removed preserves the import test consistency between venv and virtualenv.
-        # Additionally, in the above example, the pyparsing package vendored with pkg-resources is out-of-date and generates deprecation warnings.
-        # Thus it is important to remove pkg-resources to prevent system installed packages from generating deprecation warnings.
-        commands.extend(collect_uninstall(packages=['pkg-resources'], ignore_errors=True))
-        commands.extend(collect_uninstall(packages=['setuptools', 'pip']))
+    if (sanity or minimize) and bool(any(isinstance(command, PipInstall) for command in commands)):
```
Comment (nitzmahone) [inline test/lib/ansible_test/_internal/python_requirements.py]: The `bool` is redundant, is it not? IIRC `any` always returns a `bool`

File: test/lib/ansible_test/_internal/python_requirements.py [hunk 2]
```diff
@@ -207,15 +231,20 @@ def collect_requirements(
...
+        # bootstrap the managed virtual environment, which will have been created without any installed packages
+        # sanity tests which install no packages skip this step
+        commands = collect_bootstrap(python) + commands
+
+        # most infrastructure packages can be removed from sanity test virtual environments after they've been created
+        # removing them reduces the size of environments cached in containers
+        packages = list(get_venv_packages(python))
```
Comment (nitzmahone) [inline test/lib/ansible_test/_internal/python_requirements.py]: Glancing over the code, `packages_to_remove` might be a clearer name here- I saw the `packages.remove('setuptools')` below and was like, "wait, I thought we wanted to keep it?!" until I realized what was happening...

**Review by nitzmahone — COMMENTED** [review]
a couple of nits, but LGTM

File: test/lib/ansible_test/_internal/python_requirements.py (hunk 1 above)
Comment (mattclay) [reply test/lib/ansible_test/_internal/python_requirements.py]: Yes, it is. I fixed that at one point and apparently failed to include it in my final changes. I'll take care of it.

File: test/lib/ansible_test/_internal/python_requirements.py (hunk 2 above)
Comment (mattclay) [reply test/lib/ansible_test/_internal/python_requirements.py]: Agreed. I'll change this as well.

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**